ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,126 @@
+NS_OPERATORS=${NS_OPERATORS:-"openstack"}
+NS_SERVICES=${NS_SERVICES:-"openstack"}
+
+# Creates directory if does not exists
+#  Parameters:
+#  $1: directory to be created
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir -p ${OUT_DIR}
+  fi
+}
+
+# Generates a debug command tp get pods logs and description. To be used with "xargs -I {}".
+#  Parameters:
+#  $1: directory to store output logs (optional)
+function gen_pods_debug_cmd {
+  local cmd
+  if [ -z "${1}" ]; then
+    cmd=("echo 'Logs for pod: {}'; oc describe pod {} > {}-describe.log; oc logs --all-containers=true {} >> {}-containers.log")
+  else
+    create_out_dir ${1}
+    cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  fi
+  echo "${cmd[@]}"
+}
+
+# Generates a debug command to get yaml output for a resource. To be used with "xargs -I {}".
+#  Parameters:
+#  $1: directory to store output logs (optional)
+function gen_resource_yaml_debug_cmd {
+  local cmd
+  if [ -z "${1}" ]; then
+    cmd=("echo 'Getting info for {} resource'; oc get {} -o yaml > {}.yaml")
+  else
+    create_out_dir ${1}
+    cmd=("echo 'Getting info for {} resource'; oc get {} -o yaml > ${1}/{}.yaml")
+  fi
+  echo "${cmd[@]}"
+}
+
+# Get all yaml outputs from resource in "oc get all" result.
+#  Parameters:
+#  $1: namespace
+#  $2: string to be used in egrep to filter results (optional)
+function get_all_from_ns {
+  local get_all
+  local cmd
+  local dir
+  if [ -z "${2}" ]; then
+    get_all=$(oc get all -o name -n "${1}")
+  else
+    get_all=$(oc get all -o name -n "${1}" | egrep -iv "${2}")
+  fi
+
+  cmd=$(gen_resource_yaml_debug_cmd)
+  for i in ${get_all}; do
+    dir=$(echo $i | cut -d "/" -f1)
+    create_out_dir "${dir}"
+    echo ${i} | xargs -n1 -I {} sh -c "${cmd[@]}"
+  done
+}
 
 # We don't want to use OpenShift-CI build cluster namespace
 unset NAMESPACE
+oc project "${NS_OPERATORS}"
 
-oc project openstack
+pushd "${BASE_DIR}" || exit
+mkdir logs
+pushd logs || exit
 
-oc logs ceph
+oc get all > get_all.log
 
-oc get OpenStackControlPlane openstack -o json
+# api-resources - OpenStack
+CMD=$(gen_resource_yaml_debug_cmd "api-resources")
+oc api-resources | grep 'openstack' | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
+# RabbitMQCluster
+oc get -o yaml RabbitMQCluster > RabbitMQCluster.yaml
 
-oc get all
+### Pods
+# Pods - controller manager
+CMD=$(gen_pods_debug_cmd "controller-manager")
+oc get pods -n "${NS_OPERATORS}" --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
 
-oc get -o yaml MariaDB,RabbitMQCluster,KeystoneAPI,PlacementAPI,Glance,Cinder,NeutronAPI,Nova
-
-oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | xargs -n1 oc logs
+# Pods - all others
+CMD=$(gen_pods_debug_cmd "pods")
```

Comment:
Do I understand correclty that we collect all the service pod logs and then we collect again the ones in error so that we can highlight the error ones? (I'm OK with this approach)

Reply:
Yes, I'm just doing this because that logic was already here, someone added this in a previous commit. I think that might help on finding issues too.